Removes blocking overbuff, absence of insulated cap's gloves, overeffectiveness of tasers vs armor, and lack of destructible objects

TobyThorne opened the pull request:
tl;dr 
Тазеры понерфлены против брони, из блока можно контратаковать, стрелять, блокируя есвордом во второй руке теперь не так эффективно, больше разрушаемости.

- Исправлено отсутствие у .44 пробивания брони, имеющегося у всех остальных типов патронов.
- После успешного блокирования атаки в ближнем бою появляется окно в 1.5 секунды, во время которого можно контратаковать. Если атаковать, то блок отключится.
- Блокирование выстрелов при низком пойзе больше не роняет, а выбивает предмет из рук.
- Отражение выстрелов световым мечом/энергощитом, удерживаемым в off-hand теперь также тратит пойз. При низком пойзе предмет будет выбит из рук.
- Слегка усилена защита энергощитов.
- Ионка и метеоры теперь не блокируются щитами и мечами.
- Стрельба по кистям рук теперь имеет шанс выбить предмет из соответствующей руки. Шанс зависит от урона снаряда и вызываемой им боли (к примеру, бинбэги - практически идеальный выбор). Чем больше предмет, тем выше шанс, что он будет выбит из рук. Предметы размера NORMAL и выше при этом предотвращают получение урона. Неснимаемые предметы (армблядь) не выбиваются, но блокируют выстрел. Наносит урон по пойзу в 20%.
- На тазеры теперь влияет электропроводность одежды (абляционная броня теперь полностью контрит тазеры, да).
- Уменьшен рандомный шанс уронить человека стрельбой по коленям/пяткам с помощью тазера, если они защищены.
- Отрегулирована электропроводность различных элементов одежды.
- Капитанские перчатки теперь изолированы. В конце концов, они одни на станции.

Отдельный кусок работы был сделан для взаимодействия с объектами и их разрушаемости.
- Каркасы стен (wall girders):
Каркасы теперь имеют 150 ХП (300 у укреплённых), откручивание от пола больше не меняет их ХП.
Каркасы можно ломать. Оружие должно иметь как минимум 15 урона;
Каркасы можно быстро разбирать энергомечами/блейдами/энерготопорами (HELP intent);
Каркасы можно чинить с помощью сварки;
Каркасы блокируют 60% выстрелов вместо 50% (80% вместо 75% у укреплённых);
- Торговые автоматы:
Автоматы теперь имеют 100 ХП;
Автоматы можно ломать вручную (минимум 10 урона) и выстрелами;
Сломанный автомат можно починить сваркой;
- Стены:
Вместо рандомного шанса сломать стену, они теперь корректно получают урон при ударах;
Использование потайной двери в стене теперь показывает сообщение пользователю;
- Столы:
Столы теперь можно быстро ломать энергомечами и т.д. (HARM intent);
Столы теперь можно бить (HARM intent). Урон, необходимый для повреждения стола зависит от материала, но не может быть ниже 10. При ударах столы разбрасывают лежащие на них предметы в разные стороны.
Удар по столу пустой рукой (HARM intent) стукнет без урона. Бам.
- Удары по дверям и окнам теперь используют правильный кулдаун атаки;
- Кровати и стулья теперь ломаются от выстрелов;
- Для разбивания окон теперь требуется минимум 3 урона;
- Добавлен новый звук, проигрывающийся при ударах по различным объектам, в т.ч. по стенам, камерам и нестеклянным шлюзам;
- Steel shrapnel заменена на steel scrap с новым спрайтом. Шрапнель от выстрелов остаётся прежней.

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute]

Commit:
fightingupdate

- Fixed .44 ammo having no armor_penetration set up;
- After blocking an attack, you have a 1.5s window for counterattack without disabling your block mode;
- Blocking ranged attacks with shields on low poise no longer drops users down, but rathers knocks a shield out of their hands;
- Reflecting ranged attacks with eswords/eshields held in offhand slot now drains poise;
- Energy Shields' blocking properties have been increased;
- Shooting peoples' hands now have a chance to knock out the item held in respective hand. The chance is increased for more powerful ammo types, as well as for pain-inflicting types (e.g. bean bags are nearly perfect for this). Bigger items also have a higher chance to be knocked out of hands. Tiny and small items don't prevent the damage from being inflicted upon getting knocked out.

koronoshka commented:
Я ебнусь если так можно будет снять армблэйды

TobyThorne commented:
> Я ебнусь если так можно будет снять армблэйды

Отстреливаем армблейды 10 часов.
А вообще, да, смешное. Забыл про них.

Wuattt commented:
за название иссуя заплатил бы 4.20 долларов

Commit:
part2

- Unremovable items (e.g. armblades) will no longer get shot out; they will rather completely block the projectile;
- Getting an item shot out of your hands will damage your poise by 20%
- Tasers are now affected by target's clothing siemens coefficient;
- Reduced the efficiency of tasing people's knees'n'feet while they are wearing armor;
- Adjusted various clothing siemens coefficient;
- Cap's gloves are now insulated;

Commit:
crashnsmash

- Added more stuff for girders:
Girders now have 150 hp (300 for reinforced), dislodging now doesn't affect their HP.
Girders can now be damaged w/ melee weapons with force >= 15.
Girders can now be sliced down using energy swords/blades/axes (HELP intent).
Girders can now be fixed w/ welding tools.
Girders now provide 60% cover effectiveness instead of 50% (80% instead of 75% for reinforced).
- More stuff for vending machines:
Vending machines can now be damaged w/ melee weapons (force >= 10) and projectiles.
Vending machines now have 100 hp.
Broken vending machines can be fixed w/ welding tools.
- Beds/chairs stuff:
They can now be destroyed w/ projectiles.
- Walls stuff:
Instead of using random to break a wall, melee weapons now actually inflict damage.
Using hidden walls now shows a message to the user.
- Attacking doors and windows now uses correct cooldown
- Tables can now be sliced using energy swords/axes/etc. (HARM intent)
- Added a new sound for hitting objects
- Windows now require at least 3 force to get damaged

Commit:
Update girders.dm

- Fixed repair time;

TobyThorne commented:
- Каркасы стен (wall girders):
Каркасы теперь имеют 150 ХП (300 у укреплённых), откручивание от пола больше не меняет их ХП.
Каркасы можно ломать. Оружие должно иметь как минимум 15 урона;
Каркасы можно быстро разбирать энергомечами/блейдами/энерготопорами (HELP intent);
Каркасы можно чинить с помощью сварки;
Каркасы блокируют 60% выстрелов вместо 50% (80% вместо 75% у укреплённых);
- Торговые автоматы:
Автоматов теперь имеют 100 ХП;
Автоматы можно ломать вручную (минимум 10 урона) и выстрелами;
Сломанный автомат можно починить сваркой;
- Стены:
Вместо рандомного шанса сломать стену, они теперь корректно получают урон при ударах;
Использование потайной двери в стене теперь показывает сообщение пользователю;
- Удары по дверям и окнам теперь используют правильный кулдаун атаки;
- Столы теперь можно быстро ломать энергомечами и т.д. (HARM intent);
- Кровати и стулья теперь ломаются от выстрелов;
- Для разбивания окон теперь требуется минимум 3 урона;
- Добавлен новый звук, проигрывающийся при ударах по различным объектам, в т.ч. по стенам, камерам и нестеклянным шлюзам;

ArmolitskiyCarpoeb commented:
В гетто сейчас можно найти наколенники (kneepads). Вешаются на униформу как аксессуар. Может дать им какую-нибудь защиту?

TapKo4eB commented on code/modules/tables/interactions.dm:
```diff
@@ -111,7 +111,7 @@
 		return
 
-	if(istype(W, /obj/item/weapon/melee/energy/blade))
+	if(istype(W, /obj/item/weapon/melee/energy) && user.a_intent == I_HURT && W.force > 20)
```
Может быть для дальнейшего удобства сделать атаку столов на харме любыми предметами? У нас же не только энергомечами махать могут рядом со столами, и во время баттонмеша можно оружие на стол положить.
Сделать как с дверьми, урон будет наноситься только если больше урон выше трешхолда.

TobyThorne replied:
Ваще можно, но столы и так уже можно переворачивать/залезать на них. А в драке положить болон на стол - цветное...

TobyThorne replied:
Granted.

MrMagick2104 commented:
> В гетто сейчас можно найти наколенники (kneepads). Вешаются на униформу как аксессуар. Может дать им какую-нибудь защиту?

Они же просто для тактикульщиков, кто не может себе позволить плитник с наколенниками. Плюс, они есть в лоадауте. Давать им защиту - такое себе.
Наколенники эти, кстати, из разряда велосипедных. От чего они защищать-то будут?

anihillator commented:
> Наколенники эти, кстати, из разряда велосипедных. От чего они защищать-то будут?
Та какую-нибудь минимальную брут защиту на ноги, мб шанс перелома уменьшить.

MrMagick2104 commented:
> Наколенники эти, кстати, из разряда велосипедных. От чего они защищать-то будут?
Та какую-нибудь минимальную брут защиту на ноги, мб шанс перелома уменьшить.

Ну такое. Ноги обычно всё равно не в коленях ломаются. 
Ну и "минимальная брут защита" уже есть почти на всех предметах одежды в виде защиты от ударов в ближнем бою на 10%.

Commit:
unblockable projectiles

- Ion shots and meteors can't be blocked/reflected;

TobyThorne commented:
- Ионка и метеоры теперь не блокируются щитами и мечами.

Commit:
descructible_tables

- Tables can now be destroyed upon hitting them w/ HURT intent. The required force is based on the material's integrity, but cannot  be lower than 10.
- Steel shrapnel replaced with steel scrap; shrapnel spawned by bullets remains the same;
- Futher nerfed offhand esword blocking;

TobyThorne commented:
- Столы теперь можно бить (HARM intent). Урон, необходимый для повреждения стола зависит от материала, но не может быть ниже 10. При ударах столы разбрасывают лежащие на них предметы в разные стороны (можно играть в русскую рулетку, поставив материю на стол и хуяча по нему болоном).
- Steel shrapnel заменена на steel scrap с новым спрайтом. Шрапнель от выстрелов остаётся прежней.
- Дополнительно увеличен урон по пойзу от блокирования есвордом в оффхенде.

koronoshka commented:
Когда
![image](https://user-images.githubusercontent.com/58358430/83288556-36163400-a1ec-11ea-8e14-782d281d1c97.png)

MrMagick2104 commented:
> Steel shrapnel заменена на steel scrap с новым спрайтом. Шрапнель от выстрелов остаётся прежней.

А зачем?

TobyThorne commented:
> > Steel shrapnel заменена на steel scrap с новым спрайтом. Шрапнель от выстрелов остаётся прежней.
> 
> А зачем?

Для риализма и зрелищности. Когда разбиваешь какой-нибудь столик, а из него дропается заноза из 12 пикселей, как от пули - чёт такое.

Commit:
deskslam

- Clicking on a table with empty hand and HARM intent now makes you slam the table (probably for objection purposes);

TobyThorne commented:
По столам теперь можно стучать на харме.

koronoshka commented:
![Kamikadze](https://user-images.githubusercontent.com/58358430/83293942-29e2a480-a1f5-11ea-9d52-1a12f1e994d4.gif)

TapKo4eB commented on code/modules/tables/interactions.dm:
```diff
@@ -75,6 +75,28 @@
+		if(prob(50))
+			playsound(loc, 'sound/effects/deskslam.ogg', 50, 1)
+		else
+			playsound(loc, 'sound/effects/magnetclamp.ogg', 60, 1)
```
Кламп звучит как булщит при ударе по столу, поменяй или выпили.

TobyThorne replied:
выпилил nahooy, пусть пока будет только из ace attourney
искать бесплатные звуки в 2к20 - это пиздец какой-то

TapKo4eB commented:
![изображение](https://user-images.githubusercontent.com/48761132/83302442-06732600-a204-11ea-85bf-d2e37c27afbc.png)
У тебя теперь и шрапнель и скрап дропается. Бай дезигн?

TobyThorne commented:
> У тебя теперь и шрапнель и скрап дропается. Бай дезигн?

чуть не обосрался, пока пикча не загрузилась
шрапнель же от пластика, скрап только для стали (и всяких пласталей-титанов-etc) добавил

TobyThorne committed:
Update interactions.dm

TapKo4eB commented:
А ок.

Commit:
cringe

PlinHost commented on code/game/objects/structures/girders.dm:
```diff
@@ -56,50 +63,56 @@
 /obj/structure/girder/attackby(obj/item/W as obj, mob/user as mob)
 	if(isWrench(W) && state == 0)
 		if(anchored && !reinf_material)
-			playsound(src.loc, 'sound/items/Ratchet.ogg', 100, 1)
```
А куда звук пропал?

PlinHost commented on code/modules/tables/interactions.dm:
```diff
@@ -125,6 +144,33 @@
+			dam_threshhold *= 2
+		if(W.force >= dam_threshhold)
+			user.visible_message(SPAN("danger", "[user] hits \the [src] with \the [W]!"))
+			var/list/targets = list(get_step(src,dir),get_step(src,turn(dir, 45)),get_step(src,turn(dir, -45)))
+			for(var/atom/movable/A in get_turf(src))
+				var/dropchance = 50
+				if(istype(A,/obj))
+					var/obj/O = A
+					if(O.w_class)
+						dropchance = 120 - O.w_class*20 // 100% for tiny items, 80% for small items, etc.
+				if(!A.anchored && prob(dropchance))
+					spawn(0)
+						A.throw_at(pick(targets),1,1)
+			take_damage(W.force/1.5)
```
`take_damage` тоже может вызвать разброс итемов, а два разброса со spawn могут наделать смешного. Реально ли нужен `spawn(0)`?
Да и разброс по хорошему бы вынести в отдельный прок.

PlinHost commented on code/game/machinery/vending.dm:
```diff
@@ -151,6 +164,18 @@
+		return
+
+	..()
+	if(health > 0)
+		health -= damage
+		if(health <= 0)
+			set_broken(1)
```
Получение дамага нужно в отдельный прок вынести, а не копипастить.

Commit:
refactor

- some OOP stuff

Epicus7 commented:
Этот ПР закрывает платный иссуй #1130

koronoshka commented:
Сектор приз на барабане!

Commit:
Update Chemistry-Reagents-Medicine.dm

- reverted adrenaline change to prevent conflicting w/ #2408

TobyThorne committed:
Update Chemistry-Reagents-Medicine.dm

TobyThorne committed:
Update Chemistry-Reagents-Medicine.dm

Commit:
Update Chemistry-Reagents-Medicine.dm

Commit:
Update item_attack.dm

TapKo4eB commented:
Аффектит #1130 к слову.

TapKo4eB reviewed (approved):
вродь ничё не нашёл

Epicus7 reviewed (dismissed):
Чем гирдер, кровать и стена лучше миллиона других структур которые есть у нас в билде?

Костыльная обработка урона в них не нужна. Предлагаю откатить и сделать нормально отдельными ПРами. Я уже с пауками переношу зачатки адекватной системы урона по структурам.

PlinHost reviewed (approved):
2506 хорошее число

Epicus7 reviewed (commented):
Плин сказал что я дурак и нам нужно больше копипасты в билде.
